thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/DetectionResource.java

```diff
@@ -540,6 +541,41 @@ public Response createUserAnomaly(
     return Response.ok(anomaly.getId()).build();
   }
 
+  /**
+   * Toggle active/inactive for given detection
+   *
+   * @param detectionId detection config id (must exist)
+   */
+  @PUT
+  @Path("/activation/{id}")
+  @ApiOperation("Make detection active or inactive, given id")
+  public Response toggleActivation(
+      @PathParam("id") long detectionId) throws Exception {
+    Map<String, String> responseMessage = new HashMap<>();
+    Boolean toggledTo = false;
+    try {
+      DetectionConfigDTO config = this.configDAO.findById(detectionId);
+      if (config == null) {
+        throw new IllegalArgumentException(String.format("Cannot find config %d", detectionId));
+      }
+
+      // grab for logs
+      toggledTo = !config.isActive();
+      // update state
+      config.setActive(!config.isActive());
```

Comment:
config.setActive(toggledTo) is better.
Let's avoid duplicating logic, although it is very small logic.